src/test/java/io/logz/logback/LogzioLogbackAppenderTest.java

```diff
@@ -239,7 +242,35 @@ public void testMarker() throws Exception {
     }
 
     @Test
-    public void testContextReset() throws Exception {
+    public void testJsonMarkers() {
+        String token = "markersJsonToken";
+        String type = "markersJsonType";
+        String loggerName = "markersJsonTesting";
+        String markerKey = "marker";
+        int drainTimeout = 1;
+        String message1 = "Simple log line - "+random(5);
+
+        Logger testLogger = createLogger(token, type, loggerName, drainTimeout, false, false, null);
+        LogzioLogbackAppender logzioLogbackAppender =
+                (LogzioLogbackAppender)((ch.qos.logback.classic.Logger)testLogger).getAppender("LogzioLogbackAppender");
+        logzioLogbackAppender.setMarkersFormat("json");
+
+        Map logstashMarker = new HashMap();
+        logstashMarker.put("customkey_str", "value1");
+        logstashMarker.put("projectid_int", 5);
+        testLogger.info(Markers.appendEntries(logstashMarker) , message1);
```

Comment:
Are you sure that customer use only Markers.appendEntries(logstashMarker) ? As i see it may be other append functions in logstash and Markers.append with object does not work. It looks like need more research in this area.